distribution/tools/server-cli/src/test/java/org/elasticsearch/server/cli/ServerProcessTests.java

```diff
@@ -277,82 +280,119 @@ public void testLibffiEnv() throws Exception {
-            return new ArrayList<>();
-        };
-        runForeground();
+        var tmpDir = ServerProcessUtils.setupTempDir(createProcessInfo());
+        assertThat(tmpDir.toString(), Files.exists(tmpDir), is(true));
+        assertThat(tmpDir.getFileName().toString(), equalTo("elasticsearch"));
+        assertThat(tmpDir.getParent().toString(), equalTo(baseTmpDir.toString()));
     }
 
     public void testTempDirOverride() throws Exception {
         Path customTmpDir = createTempDir();
         envVars.put("ES_TMPDIR", customTmpDir.toString());
-        optionsBuilder = (args, configDir, tmpDir, envOptions) -> {
-            assertThat(tmpDir.toString(), equalTo(customTmpDir.toString()));
-            return new ArrayList<>();
-        };
+        var tmpDir = ServerProcessUtils.setupTempDir(createProcessInfo());
+        assertThat(tmpDir.toString(), equalTo(customTmpDir.toString()));
+
         processValidator = pb -> assertThat(pb.environment(), not(hasKey("ES_TMPDIR")));
         runForeground();
     }
 
-    public void testTempDirOverrideMissing() throws Exception {
+    public void testTempDirOverrideMissing() {
         Path baseDir = createTempDir();
         envVars.put("ES_TMPDIR", baseDir.resolve("dne").toString());
-        var e = expectThrows(UserException.class, () -> runForeground());
+        var e = expectThrows(UserException.class, () -> ServerProcessUtils.setupTempDir(createProcessInfo()));
         assertThat(e.exitCode, equalTo(ExitCodes.CONFIG));
         assertThat(e.getMessage(), containsString("dne] does not exist"));
     }
 
     public void testTempDirOverrideNotADirectory() throws Exception {
         Path tmpFile = createTempFile();
         envVars.put("ES_TMPDIR", tmpFile.toString());
-        var e = expectThrows(UserException.class, () -> runForeground());
+        var e = expectThrows(UserException.class, () -> ServerProcessUtils.setupTempDir(createProcessInfo()));
         assertThat(e.exitCode, equalTo(ExitCodes.CONFIG));
         assertThat(e.getMessage(), containsString("is not a directory"));
     }
 
     public void testCustomJvmOptions() throws Exception {
+        var serverArgs = createServerArgs(false, false);
+        var jvmOptionsFile = serverArgs.configDir().resolve("jvm.options");
+        var esConfigFile = serverArgs.configDir().resolve("elasticsearch.yml");
+        try {
+            Files.createDirectories(serverArgs.configDir());
+            Files.createFile(jvmOptionsFile);
+            Files.createFile(esConfigFile);
+        } catch (IOException ex) {
+            // File or directory already exists
+        }
+
         envVars.put("ES_JAVA_OPTS", "-Dmyoption=foo");
-        optionsBuilder = (args, configDir, tmpDir, envOptions) -> {
-            assertThat(envOptions, equalTo("-Dmyoption=foo"));
-            return new ArrayList<>();
-        };
+
+        var jvmOptions = JvmOptionsParser.determineJvmOptions(serverArgs, createProcessInfo(), Path.of(".'"));
```

Comment:
We already have JvmOptionsParserTests for testing actual parsing/resolution of jvm options. The point of this test should be to check that jvm options are actually passed through to the process.

Reply:
I changed the test to see that the option in ES_JAVA_OPTS are passed to the command line; if we just want to test that options are passed through, this is already covered by another test and we can remove this one. I would keep it, as it now shows that options in ES_JAVA_OPTS are read, put in jvmOptions, and also passed to the command line.

Reply:
I'm not sure it really shows that anymore. It hinges on determineJvmOptions being called. I would prefer to keep tests about ServerProcess itself, which doesn't care how the jvm options are determined.

Reply:
Fine with me. The part about setting up the correct env is covered by `testEnvCleared`, and the part about passing jvm options down to the process by `testCommandLineSysprops`